### PR TITLE
Dockerizing

### PR DESCRIPTION
**Pull request by jimbandrews**
### **1. Targeted Issue** ###

*(The first step to a good solution is knowing the problem. Please link the issue here.)*

Issue #... or
Ticket #...

https://trello.com/c/RBwhtEEA/69-implement-docker-be

### **2. Overview of Solution** ###

*(You've put in the hard work, now's your chance to tell us about it. What's your solution in a nutshell?)*

The entire backend (api, redis, and celery servers, database) can be run locally inside docker containers. This requires Docker and Docker Desktop to be installed, and simplifies running it locally to a single terminal window and (at most) 3 commands. 

### **3. Tools Used** ###

*(Share the tools, libraries, or techniques you've used in your solution.)*

Docker, Docker Compose

### **4. Testing Strategy** ###

*(No code is complete without tests. Outline what you've tested and how.)*

Follow the new instructions I added in the README for setup and teardown.

I've spun up the containers and used Postman/Insomnia to access the API endpoints. I've also connected to the database container via Postico 2.

### **5. Future Implications** ###

*(Let's anticipate potential effects of your changes. Are there any potential issues down the line?)*

Contributors will need to update their environment variables for the database connection. I had to change this to make it easier for connecting the api and celery containers to the database.

This will give new contributors a chance to familiarize themselves with Docker, which is a necessary technology in today's industry. It will also simplify setup, leading to less confusion when starting development.

### **6. Screenshots** ###

*(A picture is worth a thousand words. Include screenshots of your changes if relevant.)*

### **7. Code Reviewers** ###

*(Who else is in the loop? Mention your team or other contributors here.)*

Peri Sparr, Laura Andrews, and Christina Conley all participated in group meetings to setup and debug this feature.

---

Before you go...

- [x ]  I have linked the relevant issue
- [x ]  I have provided an overview of my solution
- [x ]  I have shared the tools I've used
- [x ]  I have outlined my testing strategy
- [x ]  I have considered future implications of my changes
- [x ]  I have included relevant screenshots
- [x ]  I have kept my team in the loop

---

### Big high-fives for your incredible pull request! We're thrilled to have you as a contributor! Our team will dig into your proposed changes and give you some feedback or merge them if they're good to go. Your contributions mean the world to us!    
Screenshot 2023-10-12 at 11 48 36 AM
Screenshot 2023-10-12 at 11 48 56 AM
Screenshot 2023-10-12 at 11 49 16 AM
Screenshot 2023-10-12 at 11 49 42 AM

### CODE REVIEW

**Review by esparr — CHANGES_REQUESTED** [review]
Very excited to have docker set up. LETS GOOOO

File: team_production_system/serializers.py
[the remark this answers is not in the document]
Comment (jimbandrews) [reply team_production_system/serializers.py:1]: File removed

File: team_production_system/urls.py
[the remark this answers is not in the document]
Comment (jimbandrews) [reply team_production_system/urls.py:1]: Changes were to make code more legible

File: team_production_system/views.py
[the remark this answers is not in the document]
Comment (jimbandrews) [reply team_production_system/views.py:1]: File removed

File: config/settings.py [hunk 1]
```diff
@@ -197,6 +205,7 @@
     'DEFAULT_AUTHENTICATION_CLASSES': (
         'rest_framework.authentication.TokenAuthentication',
     ),
+    'DEFAULT_SCHEMA_CLASS': 'drf_spectacular.openapi.AutoSchema',
```
[the remark this answers is not in the document]
Comment (jimbandrews) [reply config/settings.py]: Line deleted

File: config/settings.py [hunk 2]
```diff
@@ -62,6 +62,7 @@
     'multiselectfield',
     "team_production_system",
     "django_celery_beat",
+    'drf_spectacular',
```
[the remark this answers is not in the document]
Comment (jimbandrews) [reply config/settings.py]: Line deleted

**Review by esparr — APPROVED** [review]
Very excited to have docker set up. LETS GOOOO

**Pull request merged**